### PR TITLE
(#88) Reset the debugger and reload program

### PR DESCRIPTION
**Pull request by herrhotzenplotz**
It is a bit more difficult than anticipated.

The issue is the breakpoints. (See my notes in the code)

### CODE REVIEW

**Comment by herrhotzenplotz** [comment]
As mentioned, there is a certain complexity in reloading the breakpoints. As I implemented the actual algorithm, it wasn't as complex as it first appeared to be.

If you have any issues with the changes, please let me know.

**Comment by rexim** [comment]
@herrhotzenplotz please review my changes. Thanks!

File: src/bdb.c [hunk 1]
```diff
@@ -23,162 +23,120 @@ void usage(void)
...
+    assert(breakpoint <= state->breakpoints);
 
-    if (!state->breakpoints[addr].is_enabled)
-    {
-        fprintf(stderr, "ERR : No such breakpoint\n");
-        return;
+    state->breakpoints_size -= 1;
+    memmove(breakpoint,
+            breakpoint + 1,
+            state->breakpoints_size - (size_t) (state->breakpoints - breakpoint));
```
Comment (rexim) [inline src/bdb.c:131]: This size calculation is probably incorrect. Size should be in bytes, but in here it's most probably in amount of breakpoints.

**Comment by herrhotzenplotz** [comment]
@rexim So far, I am okay with the changes and cannot see any obvious bugs, that have not been marked as a todo.

**Comment by rexim** [comment]
@herrhotzenplotz looks good to me! :+1: Thank you so much for the contribution and the review. :heart:

**Pull request merged**